### PR TITLE
Adding more examples of using SPIs with SRF (Set Returning Functions)

### PR DESCRIPTION
**Pull request by jsaied99**
Creating more examples to show more ways to call the Spi client and interact with SRFs.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,6 +26,7 @@ members = [
     "pgx-examples/strings",
     "pgx-examples/triggers",
     "pgx-examples/versioned_so",
```
Comment (Hoverbear) [inline Cargo.toml:28]: Could you also add a change to `.github/workflows` as well? https://github.com/tcdi/pgx/blob/1a89778046547c52c714fe6885069eab5638f965/.github/workflows/tests.yml#L180-L181

File: pgx-examples/spi_srf/spi_srf.control [hunk 2]
```diff
@@ -0,0 +1,6 @@
+comment = 'spi_srf:  Created by pgx'
+default_version = '@CARGO_VERSION@'
+module_pathname = '$libdir/spi_srf'
+relocatable = false
+superuser = false
+schema = 'spi_srf'
```
Comment (Hoverbear) [inline pgx-examples/spi_srf/spi_srf.control:6]: Needs a newline at the end of this file. (Sorry, pedantic)

Comment (jsaied99) [reply pgx-examples/spi_srf/spi_srf.control]: Oh, Okay, I just added the extra line. Thanks!

File: pgx-examples/spi_srf/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
+/*
+Portions Copyright 2019-2021 ZomboDB, LLC.
+Portions Copyright 2021-2022 Technology Concepts & Design, Inc. <support@tcdi.com>
+
+All rights reserved.
+
+Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+*/
+use pgx::TableIterator;
+use pgx::*;
```
Comment (workingjubilee) [inline pgx-examples/spi_srf/src/lib.rs:10]:
```suggestion
use pgx::prelude::*;
```

File: pgx-examples/spi_srf/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+pg_module_magic!();
```
Comment (workingjubilee) [inline pgx-examples/spi_srf/src/lib.rs:12]: Note the above change also requires:
```suggestion
pgx::pg_module_magic!();
```

File: pgx-examples/spi_srf/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    fn test_calculate_human_years() {
+        let mut results: Vec<(String, i32, String, i32)> = Vec::new();
+
+        results.push(("Fido".to_string(), 3, "Labrador".to_string(), 21));
+        results.push(("Spot".to_string(), 5, "Poodle".to_string(), 35));
+        results.push(("Rover".to_string(), 7, "Golden Retriever".to_string(), 49));
+        results.push(("Snoopy".to_string(), 9, "Beagle".to_string(), 63));
+        results.push(("Lassie".to_string(), 11, "Collie".to_string(), 77));
+        results.push(("Scooby".to_string(), 13, "Labrador".to_string(), 91));
+        results.push(("Moomba".to_string(), 15, "Labrador".to_string(), 105));
```
Comment (workingjubilee) [inline pgx-examples/spi_srf/src/lib.rs:126]: Where do you remember a Moomba from? I am curious.

File: pgx-examples/spi_srf/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+    #[pg_test]
+    fn test_calculate_human_years() {
+        let mut results: Vec<(String, i32, String, i32)> = Vec::new();
+
+        results.push(("Fido".to_string(), 3, "Labrador".to_string(), 21));
+        results.push(("Spot".to_string(), 5, "Poodle".to_string(), 35));
+        results.push(("Rover".to_string(), 7, "Golden Retriever".to_string(), 49));
+        results.push(("Snoopy".to_string(), 9, "Beagle".to_string(), 63));
+        results.push(("Lassie".to_string(), 11, "Collie".to_string(), 77));
+        results.push(("Scooby".to_string(), 13, "Labrador".to_string(), 91));
```
Comment (workingjubilee) [inline pgx-examples/spi_srf/src/lib.rs:125]:
```suggestion
        results.push(("Scooby".to_string(), 13, "Great Dane".to_string(), 91));
```
scooby doo

**Review by workingjubilee — COMMENTED** [review]
These examples look fine! The only major change I really want to see is I would prefer to use the `use pgx::prelude::*;` style for all new examples going forward, even if `use pgx::*;` is technically viable. It will induce more correctness in the codebase.

File: pgx-examples/spi_srf/src/lib.rs (hunk 6 above)
Comment (jsaied99) [reply pgx-examples/spi_srf/src/lib.rs:125]: Haha lol. I know Scooby is not a Labrador. But for the sake of the filtering example, I need multiple dogs of the same breed.

File: pgx-examples/spi_srf/src/lib.rs (hunk 4 above)
Comment (jsaied99) [reply pgx-examples/spi_srf/src/lib.rs:12]: Are you referring to trying to use the namespace? Like using pgx:: ?

**Comment by workingjubilee** [comment]
Aaaand now I am done running around like a chicken with their head cut off.

This looks like an excellent additional example, thank you!

**Pull request merged**